src/lib/list.h

```diff
@@ -40,15 +51,26 @@ typedef bool (*filter_fn_t)(const void *a);
 typedef void *(*clone_fn_t)(const void *a);
 
 /**
- * Creates a new list item, store data, and inserts item in list (which can
- * be NULL). Returns created link, or NULL if failure. Created link can be
- * used as the list parameter to efficiently append new elements without
- * having to traverse the whole list to find the last one.
+ * @brief Append an element at the end of the list
+ * @param list any element of the list
+ * @param data the data the new element should keep
+ *
+ * @returns the appended elemement or NULL on failures.
+ *
+ * @note Created link can be used as the list parameter to efficiently append
+ * new elements without having to traverse the whole list to find the last one.
  */
 struct list *list_append_data(struct list *list, void *data);
 
 /**
  * @brief Prepend one element to the list.
+ * @param list any element of the list
+ * @param data the data the new element should keep
+ *
+ * @returns the prepended elemement or NULL on failures.
```

Comment:
isnt the prepended element the head of the list? Also element spelt wrong

Reply:
That's true. The prepended is always the first element of the list.